lib/galaxy/util/__init__.py

```diff
@@ -335,6 +335,17 @@ def shrink_string_by_size( value, size, join_by="..", left_larger=True, beginnin
     return value
 
 
+def parse_query_string( query_string ):
```

Comment:
Should this function be able to deal with malicious `query_string`s ? E.g.: `'foo=3&bar=4&'`, `'foo&pluto=4'`